packages/ui/react-ui-editor/src/hooks/useTextEditor.ts

```diff
@@ -45,46 +46,52 @@ export const useTextEditor = ({
   const onUpdate = useRef<() => void>();
   const [view, setView] = useState<EditorView>();
   const parentRef = useRef<HTMLDivElement>(null);
+
   useEffect(() => {
-    if (parentRef.current) {
-      // https://codemirror.net/docs/ref/#state.EditorStateConfig
-      const state = EditorState.create({
-        doc,
-        selection,
-        extensions: [
-          // TODO(burdon): Doesn't catch errors in keymap functions.
-          EditorView.exceptionSink.of((err) => {
-            log.catch(err);
-          }),
-          extensions,
-          EditorView.updateListener.of(() => {
-            onUpdate.current?.();
-          }),
-        ].filter(isNotFalsy),
-      });
-
-      // https://codemirror.net/docs/ref/#view.EditorViewConfig
-      const view = new EditorView({
-        parent: parentRef.current,
-        scrollTo,
-        state,
-        // NOTE: Uncomment to debug/monitor all transactions.
-        // https://codemirror.net/docs/ref/#view.EditorView.dispatch
-        dispatchTransactions: (trs, view) => {
-          if (debug) {
-            logChanges(trs);
-          }
-          view.update(trs);
-        },
-      });
-
-      setView(view);
+    if (!parentRef.current) {
+      return;
     }
 
+    // https://codemirror.net/docs/ref/#state.EditorStateConfig
+    const state = EditorState.create({
+      doc,
+      selection,
+      extensions: [
+        // TODO(burdon): Doesn't catch errors in keymap functions.
+        EditorView.exceptionSink.of((err) => {
+          log.catch(err);
+        }),
+        extensions,
+        EditorView.updateListener.of(() => {
+          onUpdate.current?.();
+        }),
+      ].filter(isNotFalsy),
+    });
+
+    // https://codemirror.net/docs/ref/#view.EditorViewConfig
+    const newView = new EditorView({
+      parent: parentRef.current,
+      scrollTo,
+      state,
+      // NOTE: Uncomment to debug/monitor all transactions.
+      // https://codemirror.net/docs/ref/#view.EditorView.dispatch
+      dispatchTransactions: (trs, view) => {
+        if (debug) {
+          logChanges(trs);
+        }
+        view.update(trs);
+      },
+    });
+
+    view?.destroy();
+    setView(newView);
+
     return () => {
       view?.destroy();
```

Comment:
so this should be newView.destroy?
(why i preferred having a single variable -- so that only the current scope is considered).